p2p/sync/sync.go

```diff
@@ -339,7 +331,7 @@ func (s *Service) adaptAndSanityCheckBlock(
 				coreC := p2p2core.AdaptClass(cls)
 				h, err := coreC.Hash()
 				if err != nil {
-					bodyCh <- blockBody{err: fmt.Errorf("class hash calculation error: %v", err)}
+					bodyCh <- BlockBody{Err: fmt.Errorf("class hash calculation error: %v", err)}
```

Comment:
I know this is old code but I believe this general code style is not good and I am using this comment to open up a discussion and hear what you guys @rianhughes and @infrmtcs think. 

I believe a dedicated channel for errors is requried in this case of scenarios and not having a type represent multiple things...